### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ErikStoupignan**
# Processing data in models 😄 
This branch was used to implement the following learning objectives:

- Set up associations between models. ✅ 
- Write SQL queries with Active Record. ✅ 

Continuing with the requirements of the activity, these were the **users**, **comments**, and **posts** that I introduced in my code.

image

They can be found at the following location: **` app/ db / seeds.rb`**

image

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: :posts_counter
+  has_many :comments
+  has_many :likes
+
+  def last_five_comments
+    comments.last(5)
+  end
+end
```
Comment (EricMbouwe) [inline app/models/post.rb]: - [x] Post model has a method that updates the posts counter for a user.

You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: :likes_counter
+end
```
Comment (EricMbouwe) [inline app/models/like.rb:10]: - [x] Like model has a method that updates the likes counter for a post.

As mentioned above you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_likes_counter

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @ErikStoupignan, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- Model associations ✔
- No linter errors  ✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (ErikStoupignan) [reply app/models/post.rb]: Thanks for the observation! I have already applied the requested changes.

File: app/models/like.rb (hunk 2 above)
Comment (ErikStoupignan) [reply app/models/like.rb:10]: Thanks for the observation! I have already applied the requested changes.

**Review by Ismailco — APPROVED** [review]
### Hi @ErikStoupignan, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**